case-server/src/main/java/com/powsybl/caseserver/CaseService.java

```diff
@@ -70,24 +68,67 @@ String getFormat(Path caseFile) {
 
     List<CaseInfos> getCases() {
         checkStorageInitialization();
-        try (Stream<Path> walk = Files.walk(getStorageRootDir())) {
+        try (Stream<Path> walk = Files.walk(getPublicStorageDir())) {
             return walk.filter(Files::isRegularFile)
-                    .map(file -> new CaseInfos(file.getFileName().toString(), getFormat(file)))
+                    .map(file -> new CaseInfos(file.getFileName().toString(), getFormat(file), UUID.fromString(file.getParent().getFileName().toString())))
                     .collect(Collectors.toList());
         } catch (IOException e) {
             throw new UncheckedIOException(e);
         }
     }
 
-    Path getCaseFile(String caseName) {
-        validateCaseName(caseName);
-        return getStorageRootDir().resolve(caseName);
+    public Path getCaseFile(UUID caseUuid) {
+        Path publicCaseDirectory = getPublicStorageDir();
+        Path privateCaseDirectory = getPrivateStorageDir();
+        Path caseFile = walkCaseDirectory(publicCaseDirectory.resolve(caseUuid.toString()));
+        if (caseFile != null) {
+            return caseFile;
+        }
+        caseFile = walkCaseDirectory(privateCaseDirectory.resolve(caseUuid.toString()));
+        if (caseFile != null) {
+            return caseFile;
+        }
+        throw CaseException.createDirectoryNotFound(caseUuid);
+    }
+
+    Path getCaseDirectory(UUID caseUuid) {
+        Path publicCaseDirectory = getPublicStorageDir();
+        Path privateCaseDirectory = getPrivateStorageDir();
+        Path caseFile = publicCaseDirectory.resolve(caseUuid.toString());
+        if (Files.exists(caseFile) && Files.isDirectory(caseFile)) {
+            return caseFile;
+        }
+        caseFile = privateCaseDirectory.resolve(caseUuid.toString());
+        if (Files.exists(caseFile) && Files.isDirectory(caseFile)) {
+            return caseFile;
+        }
+        throw CaseException.createDirectoryNotFound(caseUuid);
+    }
+
+    public Path walkCaseDirectory(Path caseDirectory) {
+        if (Files.exists(caseDirectory) && Files.isDirectory(caseDirectory)) {
+            try (Stream<Path> pathStream = Files.walk(caseDirectory)) {
+                List<Path> list = pathStream.filter(file -> !Files.isDirectory(file)).collect(Collectors.toList());
+                if (list.isEmpty()) {
+                    throw CaseException.createDirectoryEmpty(caseDirectory);
+                }
+                return list.get(0);
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
+            }
+        }
+        return null;
     }
 
-    Optional<byte[]> getCaseBytes(String caseName) {
+    Optional<byte[]> getCaseBytes(UUID caseUuid) {
         checkStorageInitialization();
 
-        Path caseFile = getCaseFile(caseName);
+        Path caseFile;
+        try {
+            caseFile = getCaseFile(caseUuid);
+        } catch (CaseException e) {
+            return Optional.empty();
```

Comment:
I think it's a bad design to use exceptions inside a single class like this. Exceptions are usefull for going through the callstack between two distant pieces of code. Inside one class, simpler control flows is better. Try to remove the exception ?